homeassistant/components/sense/__init__.py

```diff
@@ -91,11 +92,14 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     except SENSE_EXCEPTIONS as err:
         raise ConfigEntryNotReady(str(err) or "Error during realtime update") from err
 
+    async def _update_func():
+        await gateway.update_trend_data(dt_util.now())
```

Comment:
Why we need to tell the API about Home Assistant's local time?

Reply:
The library needs to send a date to the API with the current date which could be wrong if the server HA is running on isn't set to the same timezone.  If the HA local time is sent, then it matches what the user has set which is likely to be the correct timezone

Reply:
Then we should use UTC here, not local time + guessing.

Reply:
The Sense server records the data in the user's timezone.  For example if the user is in EST, then it records it in EST and  sending the UTC date means that any request sent at 10PM would request data from the next day (which returns 0).

Reply:
OK, so the user has configured the timezone in the Sense service? Can we get access to the configures timezone via the API?

Reply:
It's exposed in the API as a string, however the library would need to be extended to read it

Reply:
OK, then we should do that; timezone setting in Home Assistant is not straight forwards since we have both the server's timezone and the timezone configured in Home Assistant which may not be identical.

Reply:
I'd imagine the Home Assistant timezone would match the Sense timezone.  Server timezone is what's being eliminated with this change